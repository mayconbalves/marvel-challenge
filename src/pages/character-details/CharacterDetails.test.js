import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import CharacterDetails from './'

describe('<CharacterDetails />', () => {
  const mockStore = configureStore([thunk])
  let store
  store = mockStore({
    charactersReducer: { characters: [] }
  })

  it('should render link logo', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CharacterDetails />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('link', { name: /logo/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render logo image', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CharacterDetails />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('img', { name: /marvel/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render title when have comics', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CharacterDetails />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('heading', { name: /Últimos lançamentos/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render title without comics', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CharacterDetails />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('heading', { name: /Not found comics:/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
