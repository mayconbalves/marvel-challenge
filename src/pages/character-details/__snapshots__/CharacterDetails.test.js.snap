// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterDetails /> should render link logo 1`] = `
<div
  class="sc-bdfBwQ jDLbtJ"
  color="#e7f6e7"
>
  <div
    class="sc-iBPRYJ gKULRa"
  >
    <div
      class="sc-dlfnbm cMUrvE"
    >
      <div
        class="sc-eCssSg cLHnhZ"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="sc-hKgILt eWTPgb"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="sc-jSgupP fgKrOJ"
      >
        <input
          class="sc-gsTCUz jhNQGZ"
          name="hero"
          placeholder="Procure seu herói"
          type="text"
          value=""
          width="80%"
        />
      </div>
    </div>
    <div
      class="sc-gKsewC bOueue"
    >
      <div
        class="sc-fubCfw cFiCdb"
      >
        <div
          class="sc-pFZIQ jnkQqj"
        >
          <h1 />
          <img
            alt="favoritar"
            src="empty_heart.png"
          />
        </div>
        <p
          class="sc-jrAGrp dULkDb"
        >
          No description
        </p>
      </div>
      <div
        class="sc-kEjbxe izSWIv"
      >
        <img
          src="undefined.undefined"
        />
      </div>
    </div>
    <div>
      <h1>
        Últimos lançamentos
      </h1>
      <div
        class="sc-iqHYGH xvFmK"
      >
        <div
          class="sc-crrsfI jDHbbv"
        >
          <h2>
            Not found comics:
          </h2>
          <img
            alt="marvel"
            src="not-found.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CharacterDetails /> should render logo image 1`] = `
<div
  class="sc-bdfBwQ jDLbtJ"
  color="#e7f6e7"
>
  <div
    class="sc-iBPRYJ gKULRa"
  >
    <div
      class="sc-dlfnbm cMUrvE"
    >
      <div
        class="sc-eCssSg cLHnhZ"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="sc-hKgILt eWTPgb"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="sc-jSgupP fgKrOJ"
      >
        <input
          class="sc-gsTCUz jhNQGZ"
          name="hero"
          placeholder="Procure seu herói"
          type="text"
          value=""
          width="80%"
        />
      </div>
    </div>
    <div
      class="sc-gKsewC bOueue"
    >
      <div
        class="sc-fubCfw cFiCdb"
      >
        <div
          class="sc-pFZIQ jnkQqj"
        >
          <h1 />
          <img
            alt="favoritar"
            src="empty_heart.png"
          />
        </div>
        <p
          class="sc-jrAGrp dULkDb"
        >
          No description
        </p>
      </div>
      <div
        class="sc-kEjbxe izSWIv"
      >
        <img
          src="undefined.undefined"
        />
      </div>
    </div>
    <div>
      <h1>
        Últimos lançamentos
      </h1>
      <div
        class="sc-iqHYGH xvFmK"
      >
        <div
          class="sc-crrsfI jDHbbv"
        >
          <h2>
            Not found comics:
          </h2>
          <img
            alt="marvel"
            src="not-found.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CharacterDetails /> should render title when have comics 1`] = `
<div
  class="sc-bdfBwQ jDLbtJ"
  color="#e7f6e7"
>
  <div
    class="sc-iBPRYJ gKULRa"
  >
    <div
      class="sc-dlfnbm cMUrvE"
    >
      <div
        class="sc-eCssSg cLHnhZ"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="sc-hKgILt eWTPgb"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="sc-jSgupP fgKrOJ"
      >
        <input
          class="sc-gsTCUz jhNQGZ"
          name="hero"
          placeholder="Procure seu herói"
          type="text"
          value=""
          width="80%"
        />
      </div>
    </div>
    <div
      class="sc-gKsewC bOueue"
    >
      <div
        class="sc-fubCfw cFiCdb"
      >
        <div
          class="sc-pFZIQ jnkQqj"
        >
          <h1 />
          <img
            alt="favoritar"
            src="empty_heart.png"
          />
        </div>
        <p
          class="sc-jrAGrp dULkDb"
        >
          No description
        </p>
      </div>
      <div
        class="sc-kEjbxe izSWIv"
      >
        <img
          src="undefined.undefined"
        />
      </div>
    </div>
    <div>
      <h1>
        Últimos lançamentos
      </h1>
      <div
        class="sc-iqHYGH xvFmK"
      >
        <div
          class="sc-crrsfI jDHbbv"
        >
          <h2>
            Not found comics:
          </h2>
          <img
            alt="marvel"
            src="not-found.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CharacterDetails /> should render title without comics 1`] = `
<div
  class="sc-bdfBwQ jDLbtJ"
  color="#e7f6e7"
>
  <div
    class="sc-iBPRYJ gKULRa"
  >
    <div
      class="sc-dlfnbm cMUrvE"
    >
      <div
        class="sc-eCssSg cLHnhZ"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="sc-hKgILt eWTPgb"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="sc-jSgupP fgKrOJ"
      >
        <input
          class="sc-gsTCUz jhNQGZ"
          name="hero"
          placeholder="Procure seu herói"
          type="text"
          value=""
          width="80%"
        />
      </div>
    </div>
    <div
      class="sc-gKsewC bOueue"
    >
      <div
        class="sc-fubCfw cFiCdb"
      >
        <div
          class="sc-pFZIQ jnkQqj"
        >
          <h1 />
          <img
            alt="favoritar"
            src="empty_heart.png"
          />
        </div>
        <p
          class="sc-jrAGrp dULkDb"
        >
          No description
        </p>
      </div>
      <div
        class="sc-kEjbxe izSWIv"
      >
        <img
          src="undefined.undefined"
        />
      </div>
    </div>
    <div>
      <h1>
        Últimos lançamentos
      </h1>
      <div
        class="sc-iqHYGH xvFmK"
      >
        <div
          class="sc-crrsfI jDHbbv"
        >
          <h2>
            Not found comics:
          </h2>
          <img
            alt="marvel"
            src="not-found.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
